### PR TITLE
Test client side rendering endpoint for pairings

### PR DESCRIPTION
**Pull request by haradan**

### CODE REVIEW

**Review by plural — CHANGES_REQUESTED** [review]
Are you able to hide the random seed changes more?  Perhaps a helper that these use without passing it around as much and that you can replace for the tests?

File: spec/requests/rounds_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+      "corp_id" => { "faction" => nil, "name" => nil },
+      "runner_id" => { "faction" => nil, "name" => nil },
+      "side_label" => nil
+    }
+  end
+
+  def bye_player
+    { "corp_id" => nil, "name_with_pronouns" => "(Bye)", "runner_id" => nil, "side_label" => nil }
+  end
+end
```
[the remark this answers is not in the document]
Comment (haradan) [reply spec/requests/rounds_controller_spec.rb:175]: You mean newlines at the end of the file? Will do.

File: app/controllers/rounds_controller.rb [hunk 2]
```diff
@@ -22,7 +22,6 @@ def pairings_data
       registrations: [player: [:user, :corp_identity_ref, :runner_identity_ref]]
     )
     render json: {
-      tournament_id: @tournament.id,
```
[the remark this answers is not in the document]
Comment (haradan) [reply app/controllers/rounds_controller.rb:25]: Yes, it's not used.

**Comment by haradan** [comment]
> Are you able to hide the random seed changes more? Perhaps a helper that these use without passing it around as much and that you can replace for the tests?

I don't know what you'd use for this in Rails, and I'm not sure I'd want to hide it. It took some digging to find out where the randomness was coming from. It sounds like helpers are intended to be used in views, and are occasionally made available in controllers?

The level of indirection makes it awkward to pass it around everywhere, but maybe the indirection is the problem. It might be better to instantiate the pairer directly rather than calling pair on the round or stage?

**Comment by plural** [comment]
> > Are you able to hide the random seed changes more? Perhaps a helper that these use without passing it around as much and that you can replace for the tests?
> 
> I don't know what you'd use for this in Rails, and I'm not sure I'd want to hide it. It took some digging to find out where the randomness was coming from. It sounds like helpers are intended to be used in views, and are occasionally made available in controllers?
> 
> The level of indirection makes it awkward to pass it around everywhere, but maybe the indirection is the problem. It might be better to instantiate the pairer directly rather than calling pair on the round or stage?

i think that is worth a shot, yeah.  I think you can leave it for a separate PR, though, if you want.

**Pull request merged**